custom/enikshay/case_utils.py

```diff
@@ -169,44 +172,52 @@ def get_all_episode_cases_from_person(domain, person_case_id):
     return episode_cases
 
 
-def get_open_episode_case_from_occurrence(domain, occurrence_case_id):
+def get_open_active_dstb_episode_case_from_occurrence(domain, occurrence_case_id):
     """
-    Gets the first open 'episode' case for the occurrence
+    In eNikshay public sector app
+    Gets the first open DSTB 'episode' case for the occurrence
 
     Assumes the following case structure:
     Occurrence <--ext-- Episode
 
     """
     case_accessor = CaseAccessors(domain)
     episode_cases = case_accessor.get_reverse_indexed_cases([occurrence_case_id])
-    open_episode_cases = [case for case in episode_cases
-                          if not case.closed and case.type == CASE_TYPE_EPISODE and
-                          case.dynamic_case_properties().get('episode_type') == "confirmed_tb"]
-    if open_episode_cases:
-        return open_episode_cases[0]
+    open_active_episode_cases = [case for case in episode_cases
+                                 if not case.closed and
+                                 case.type == CASE_TYPE_EPISODE and
+                                 case.get_case_property().get('episode_type') == "confirmed_tb" and
+                                 case.get_case_property('is_active') == 'true'
+                                 ]
+    if len(open_active_episode_cases) > 1:
+        raise EnikshayBadAppState("Multiple open active episode cases found for occurrence")
+
+    if open_active_episode_cases:
+        return open_active_episode_cases[0]
     else:
         raise ENikshayCaseNotFound(
             "Occurrence with id: {} exists but has no open episode cases".format(occurrence_case_id)
         )
 
 
-def get_open_drtb_hiv_case_from_episode(domain, episode_case_id):
+def get_open_drtb_hiv_case_from_occurrence(domain, occurrence_case_id):
     """
-    Gets the first open 'drtb-hiv-referral' case for the episode
-
-    Assumes the following case structure:
-    episode <--ext-- drtb-hiv-referral
+    - Get the occurrence case from the episode case
+    - Get an open secondary_owner case (extension of occurrence) where secondary_owner_type = drtb-hiv"
     """
     case_accessor = CaseAccessors(domain)
-    open_drtb_cases = [
-        case for case in case_accessor.get_reverse_indexed_cases([episode_case_id])
-        if not case.closed and case.type == CASE_TYPE_DRTB_HIV_REFERRAL
+    open_drtb_hivsecondary_owner_case = [
+        case for case in case_accessor.get_reverse_indexed_cases([occurrence_case_id])
+        if not case.closed and
+        case.type == CASE_TYPE_SECONDARY_OWNER and
+        case.get_case_property('secondary_owner_type') == 'drtb-hiv'
     ]
-    if open_drtb_cases:
-        return open_drtb_cases[0]
+    if open_drtb_hivsecondary_owner_case:
+        return open_drtb_hivsecondary_owner_case[0]
```

Comment:
what if there is more than one open one here? Is that an issue?

Reply:
@sheelio

Reply:
@sheelio please confirm.

Reply:
There should only ever be one open secondary owner case that is for the DRTB HIV coordinator. 

(IF there are multiple, we may want to have a reconciliation that handles this).